File: fragment-manager/src/main/java/com/bajicdusko/fragmentmanager/SFMActivity.java
package com.bajicdusko.fragmentmanager;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

/**
 * Created by Dusko Bajic on 24.06.17.
 * GitHub @bajicdusko
 */

public abstract class SFMActivity extends AppCompatActivity {

  protected abstract int getFrameLayoutContainerId();

  public SimpleFragmentManager simpleFragmentManager;

  @Override protected void onCreate(@Nullable Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    simpleFragmentManager = new SimpleFragmentManager(getSupportFragmentManager(), getFrameLayoutContainerId());
  }

  @Override protected void onSaveInstanceState(Bundle outState) {
    super.onSaveInstanceState(outState);
    simpleFragmentManager.onSaveInstanceState(outState);
  }

  @Override protected void onRestoreInstanceState(Bundle savedInstanceState) {
    super.onRestoreInstanceState(savedInstanceState);
    simpleFragmentManager.onRestoreInstanceState(savedInstanceState);
  }

  protected void onSFMBackPressed() {
    if (!simpleFragmentManager.onBackPressed()) {
      finish();
    }
  }

  @Override public void onBackPressed() {
    onSFMBackPressed();
  }
}

File: kotlin-fragment-manager/src/main/java/com/bajicdusko/kotlinfragmentmanager/SFMActivity.kt
package com.bajicdusko.kotlinfragmentmanager

import android.os.Bundle
import android.support.v7.app.AppCompatActivity

/**
 * Created by Dusko Bajic on 14.06.17.
 * GitHub @bajicdusko
 */
abstract class SFMActivity : AppCompatActivity() {

    /**
     * Override this method and return the resource id of the fragment container layout.
     */
    abstract fun getFrameLayoutContainerId(): Int

    val simpleFragmentManager by lazy { SimpleFragmentManager(supportFragmentManager, getFrameLayoutContainerId()) }

    override fun onSaveInstanceState(outState: Bundle?) {
        super.onSaveInstanceState(outState)
        simpleFragmentManager.onSaveInstanceState(outState)
    }

    override fun onRestoreInstanceState(savedInstanceState: Bundle?) {
        super.onRestoreInstanceState(savedInstanceState)
        simpleFragmentManager.onRestoreInstanceState(savedInstanceState)
    }

    protected fun onSFMBackPressed() {
        if (!simpleFragmentManager.onBackPressed()) {
            finish()
        }
    }

    override fun onBackPressed() {
        onSFMBackPressed()
    }
}